cmd/clusterctl/client/cluster/upgrader.go

```diff
@@ -44,10 +44,10 @@ type ProviderUpgrader interface {
 	Plan() ([]UpgradePlan, error)
 
 	// ApplyPlan executes an upgrade following an UpgradePlan generated by clusterctl.
-	ApplyPlan(clusterAPIVersion string) error
+	ApplyPlan(opts InstallOptions, clusterAPIVersion string) error
```

Comment:
Should this here and below be UpgradeOptions? I think we should not re-use the options struct from the ProviderInstaller

Reply:
Yep, makes sense. I will make the change.

Reply:
Done.